### PR TITLE
feat: Binance web3 wallet integration

### PR DESCRIPTION
**Pull request by chefjackson**
### Summary
🌐🎨🗑️

This pull request adds support for Binance Wallet Connect integration to the web app, allowing users to connect their Binance wallets to the app. It also adds the `tailwindcss` package to the project, enabling the use of the Tailwind CSS framework for styling. It modifies the `useEagerConnect` hook, the `pnpm-lock.yaml` file, and the `apps/web/package.json` file to implement these features.

> _`useEagerConnect`_
> _Binance Wallet joins the web_
> _Autumn of wagmi_

### Walkthrough
*  Add Binance Wallet Connect integration by importing and using `@binance/w3w-utils` and `@binance/w3w-wagmi-connector` packages ([link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-14b60f636e1a2b0061da57aaf231cb1ed15a5dc0c592425ed82e58fec95d42d8R37-R38), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-398522038f0136ce145af979005520e2298abebe07ebb5e4a6bd8e00ccbbe9d6R3-R5), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-398522038f0136ce145af979005520e2298abebe07ebb5e4a6bd8e00ccbbe9d6R12-R26), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-398522038f0136ce145af979005520e2298abebe07ebb5e4a6bd8e00ccbbe9d6L21-R42), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR641-R646), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL649-R655), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL5519-R5656), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL5526), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR3642-R3767), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR16397-R16400), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR19257-R19260))
*  Remove unused `@wagmi/core` and `abitype` packages and their dependencies ([link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL649-R655), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL5519-R5656), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL5526), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11260-L11282))
*  Add `tailwindcss` package and its dependencies to enable the use of the Tailwind CSS framework ([link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-14b60f636e1a2b0061da57aaf231cb1ed15a5dc0c592425ed82e58fec95d42d8R37-R38), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR641-R646), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR2270-R2274), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR13840-R13844), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR15333-R15336), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR15374-R15377), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR20955-R20959), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR21508-R21529), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL21397-R21577), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR22536-R22541), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR24228-R24258), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR26615-R26619))
*  Remove `dev: true` flags from several packages that are also used by `tailwindcss` for production purposes ([link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL12538-L12546), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL12806), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL12844), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17074), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL20593), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL21287), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL21329), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL21429), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL23922), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL24279), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL24285), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL24489))
*  Downgrade `@types/node` dependency of `vite` package to match the version required by `tailwindcss` ([link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL10978-R11114), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL19649-R19776), [link](https://github.com/pancakeswap/pancake-frontend/pull/8558/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL25506-R25715))

### CODE REVIEW

**Comment by chefjackson** [comment]
@SocketSecurity ignore tailwindcss@3.3.6

File: apps/web/src/hooks/useEagerConnect.ts [hunk 1]
```diff
@@ -16,6 +18,21 @@ const useEagerConnect = () => {
       return
     }
 
+    if (typeof window !== 'undefined' && isInBinance()) {
```
Comment (chef-jojo) [inline apps/web/src/hooks/useEagerConnect.ts:21]: can just use 
```
export const isInBinance = () => {
  try {
    return (window as any)?.ethereum?.isBinance === true
  } catch (error) {
    return false
  }
}

```

File: apps/web/src/hooks/useEagerConnect.ts [hunk 2]
```diff
@@ -16,6 +18,21 @@ const useEagerConnect = () => {
       return
     }
 
+    if (isInBinance()) {
+      import('@binance/w3w-wagmi-connector').then(({ getWagmiConnector }) => {
```
Comment (chef-jojo) [inline apps/web/src/hooks/useEagerConnect.ts:22]: we can reuse the connector from `connectors`?

**Comment by ChefMomota** [comment]
LGTM

**Review by chef-jojo — APPROVED** [review]

**Pull request merged**